audbcards/core/dataset.py

```diff
@@ -26,34 +26,38 @@ def __init__(
             self,
             name: str,
             version: str,
+            cache_root: str = None,
     ):
         self.name = name
         self.version = version
         self.repository = audb.repository(name, version)
+        self.cache_root = audeer.path(
+            cache_root or audb.default_cache_root(shared=False)
+        )
```

Comment:
This is not a good idea as we also use the cache to store other (non `audb` related) files there.
I would propose you add two arguments to `Dataset.__init__()`:
```python
            cache_root: str = './cache',
            audb_cache_root: str = None,
```
then you can replace the code here with

```suggestion
        self.cache_root = audeer.path(cache_root)
        self.cache_root = cache_root or audb.default_cache_root()
```
and adjust the remaining code accordingly.

Reply:
I also wasn't sure if this is a good idea, I adjusted the code to your suggestion. We still have a BUILD directory that is hard-coded, does it make sense to also add it as an argument ? But maybe in a separate PR

Reply:
The BUILD directory is slightly more complicated as it is set when executing `sphinx` to build the HTML pages. So we need to get it from there. We have already a related issue with https://github.com/audeering/audbcards/issues/2.

Reply:
Makes sense so we can skip it for this PR, but what about the `cache_root` variable, I don't see any use for it so maybe we shouldn't keep unless you think we might require it in the future ?

Reply:
You are right, I thought the usage in `player` was something different than the `audb` cache, but this is not the case. So yes, I would remove it.

What is a little bit critical is that we clean up parts of the cache (basically to save space when storing the cache inside a CI cache) at https://github.com/audeering/audbcards/blob/ce589ac917211974691f6c93416a1bdd6b2e6e50/audbcards/core/dataset.py#L47-L56

So maybe we shouldn't use the default `audb_cache_root` at all?
I would propose we just use `cache_root = './cache` as argument and state that this folder will be used as `audb` cache, but will also be managed/cleaned by `audbcards`?

Reply:
Aren't the CI jobs not supposed to be persistent anyway ? Meaning we will always be starting from fresh cache which contains only the current version (so basically just the .pkl and .csv files as well as one `wav` file which is the example file).

Also do you think there will be a case where we will end up having multiple versions of the database inside the cache inside a CI job ?

Because the reason why I thought it'd be nice to use `audb_cache_root` is in the case of large databases, however when writing this comment I realised that even for large databases it shouldn't be a problem as we're only downloading one audio file and metadata which isn't massive.

So to summarise in a clean way what I said:

1. Do we need to clean the cache in the first place when it comes to CI jobs? 
2. Do we need to use `audb_cache` to download one audio file and metadata, or is it enough if we make `audbcards` have its own cache folder which would be implemented the same way as `audb` cache. (In the case where the creation of the data card will happen on a separate machine for example so not necessary the same machine that downloaded the full dataset, this wouldn't count as redundancy. If the same machine is used then it's smarter to fetch that example file and the meta for `audb_cache`)

Reply:
> Or do you think there will be a case where we will end up having multiple versions of the database inside the cache inside a CI job ?

Yes, compare https://gitlab.audeering.com/data/data.pp.audeering.com/-/blob/main/.gitlab-ci.yml.
But I agree that cleaning the cache should then maybe handled in the project, and not inside `audbcards`.

> ... metadata which isn't massive.

It depends if you have >1000 databases and some have >1.000.000 it can still take some time to download all the dependency files and load them from CSV (the second time it will be loaded from PKL files, which is much faster).

To continue here, I would propose we first leave in the clean up code and we need to figure out how easily it can be done outside of `audbcards`. I would go with one argument here, name it `cache_root` and set its default value to `'./cache'`.

In another pull request we might then change this and support for `None` and maybe even change the default value to `None`.

Reply:
Alright, I've revert it to have only one argument as suggested.

Reply:
Cool, thanks.